apilib/service.py

```diff
@@ -101,23 +101,37 @@ def foo(self, foo_request):
             return FooResponse(...)
     '''
 
-    def invoke(self, method_name, request):
+    def invoke(self, method_name, request, already_validated=False):
```

Comment:
How about, `do_validation=True`? Just a little clearer from a "do, don't do" perspective. And then add a comment that validation is unnecessary when this is being invoked from `invoke_with_json` which does validation at deserialization time.

Reply:
Cool, I like that better!